snapcraft/hooks/configure

```
@@ -33,9 +33,12 @@ get_bool() {
     return
 }
 
-# Don't fail if the mount namespace isn't properly setup yet
+# Try to fix an improperly setup mount namespace
 if [ ! -e /run/snapd-snap.socket ]; then
-    exit 0
```

I wonder if we should just fail entirely, im not sure when the `configure` hook is called and why the snap socket shouldnt always be available.

---

So you'd like the old behaviour of exiting on missing socket but with a `rc=1`?

---

I'd like to try it, and see if it works, as otherwise it feels like it could be silently masking issues.

---

> if we should just fail entirely, im not sure when the `configure` hook is called

I think that's a main question here. We probably need to consult with snapd code before changing this. I believe there was a reason for this silent masking, for instance, it can be that the `configure` hook is being called a few times during refresh on a different stages (where mount namespace is not yet ready). By returning an error we can break that.